disnake/interactions/base.py

```diff
@@ -1017,15 +1013,15 @@ async def autocomplete(
         if self._responded:
             raise InteractionResponded(self._parent)
 
-        data = {}
+        choices_data: List[ApplicationCommandOptionChoicePayload]
         if not choices:
-            data["choices"] = []
+            choices_data = []
         elif isinstance(choices, Mapping):
-            data["choices"] = [{"name": n, "value": v} for n, v in choices.items()]
+            choices_data = [{"name": n, "value": v} for n, v in choices.items()]
         elif isinstance(choices, Iterable) and not isinstance(choices[0], OptionChoice):
-            data["choices"] = [{"name": n, "value": n} for n in choices]
+            choices_data = [{"name": n, "value": n} for n in choices]  # type: ignore
         else:
-            data["choices"] = [c.to_dict() for c in choices]  # type: ignore
+            choices_data = [c.to_dict() for c in choices]  # type: ignore
```

Comment:
how about doing it after
```py
def _parse_command_choice(choices: Choices) -> t.List[hikari.CommandChoice]:
    """Parse an arbitrary choice into a list of CommandChoice"""
    if isinstance(choices, t.Mapping):
        return [hikari.CommandChoice(name=str(name), value=value) for name, value in choices.items()]

    cc = []
    for value in choices:
        if not isinstance(value, hikari.CommandChoice):
            value = hikari.CommandChoice(name=str(value), value=value)
        cc.append(value)

    return cc
```

Reply:
yea, I mostly wanted to get the typing out of the way first, but that part needs a rework.
Should be fixed now